gradle.properties

```
@@ -5,5 +5,8 @@ dbDatarepoChangesetFile=src/main/resources/db/changelog.xml
 dbStairwayUri=jdbc:postgresql://127.0.0.1:5432/stairway
 dbStairwayUsername=drmanager
 dbStairwayPassword=drpasswd
+# Increasing max heap size from default of 512m:
+# https://docs.gradle.org/current/userguide/config_gradle.html#sec:configuring_jvm_memory
+org.gradle.jvmargs=-Xms512m -Xmx1g "-XX:MaxMetaspaceSize=384m"
```

```suggestion
org.gradle.jvmargs=-Xms512m -Xmx1g "-XX:MaxMetaspaceSize=384m"
```

Is the `MaxMetaspaceSize` something we should explicitly set? The other arguments max sense to me.

---

Good question -- yes, we should explicitly set this from what I found in my research.  (The below snippet reflects an earlier default value for this default but otherwise remains relevant to us.)

https://community.sonarsource.com/t/gradle-daemon-crashed-when-building-with-sonarqube/58744/6

>One of the key defaults set by Gradle is `-XX:MaxMetaspaceSize=256m`, which is required because [this value is unbounded on the JDK by default](https://docs.oracle.com/en/java/javase/17/gctuning/other-considerations.html#GUID-B29C9153-3530-4C15-9154-E74F44E3DAD9), allowing the Gradle daemon to consume more and more native memory until it crashes.
>
>This means that a user setting `org.gradle.jvmargs=-Xmx4g` can result in daemon crashes, unless the user knows that they must also add a `-XX:MaxMetaspaceSize` setting.

---

If this was only needed for sonar, why not set it on the command line using `GRADLE_OPTS` for the specific invocation of sonar in our GitHub workflows? Setting it generally means that it will affect running gradles, including local ones, which don't need this extra space as far as I can tell.

---

I unfortunately did not save the heap dumps or the logs, but I had experienced similar (but not identical) heap space issues locally when running unit tests last week.  My preference is to start with this property override but please feel free to PR a different approach if you'd like to noodle on it further.